Merge Team Roles & Structures docs into a single unified source of structure :) (#869)
ipfs/team-mgmt: TEAMS_ROLES_STRUCTURES.md

    - `Efforts to Support Specific Uses:`
      - [**Integration with Web Browsers**](#integration-with-web-browsers)
      - [**Dynamic Data and Capabilities**](#dynamic-data-and-capabilities)
      - [**Decentralized Data Stewardship**](#decentralized-data-stewardship)
      - [**Local Offline Collaboration**](#local-offline-collaboration-locol)
  - [**Research** Groups](#research-groups)

## Description

The IPFS Project is developed by multiple teams from multiple organizations and both intermittent and continuous invidividual contributors. A structure was developed (and continues to evolve) around Working & Research Groups, OKRs, and PM best practices that were designed to support the already large and growing group of people to coordinate among each other at a global scale.

This document presents the the existing teams that emerged and that led to the many of the sucesses within the IPFS project, their structure(namely: Working Groups & Research Groups) and a definition of roles existing in the IPFS org. At the end of this document, you will find pointers to Teams and Projects that have grown to become their own entities, namely: IPLD, libp2p and Multiformats.

> ## Team Structures

Author: @momack2 probably it is time to describe also what a Tiger Team is given that those have become more and more used.

Reviewer: Added!

### Working Group Captain

The Working Group Captain is a champion for the Product and/or Focus Area. They take the lead on writing or guiding the conversation specs, documentation and other artifacts to support the team. The Captain is also the gatekeeper of the Working Group Roadmap and accumlator of the Working Group Knowledge, guiding the group to made good decisions.

Responsibilities include:
- A champion/tech lead for the focus area
- Guide and Gatekeeper of the Working Group Roadmap
- Decision disambiguator for the technical matters of the Working Group

### Working Group Technical Project Manager (TPM)

The Technical Project Manager (sometimes referenced as Project Manager, Program Manager, and cat herder) is a team enabler. They own the Quarterly Planning process including OKRs and Retrospectives. They ensure that the coordination strategy the WG selected is well executed (weekly syncs, taking notes, communicating needs to other WGs).

The Technical Project Manager should also have a role on contributing to the projects the Working Group is tackling and have an understanding of the technical challenges and needs from the team.

> It happens that some working groups are ok with having the same person playing the Captain + Technical Project Manager role together, it works for when the Working Group is nascent or the team is small.

Reviewer 2: ```suggestion
It happens that some working groups have the same person playing both the Captain and Technical Project Manager role. This should be limited to situations where the Working Group is nascent or the team is small.
```
